root/etc/e-smith/events/actions/nethserver-nextcloud-conf

```diff
@@ -8,23 +12,90 @@ function OCC
     TERM=dumb runuser -s /bin/bash apache -c "php -dmemory_limit=512M ./occ $params"
 }
 
+exitOnError () {
+    echo $1
+    OCC "maintenance:mode --off"
+    exit 1
+}
+
 password=`perl -e "use NethServer::Password; print NethServer::Password::store('nextcloud');"`
 
-# initialize grants mysql nextcloud database
-/usr/bin/mysql --defaults-file=/root/.my.cnf -e "CREATE DATABASE IF NOT EXISTS nextcloud;"
-/usr/bin/mysql --defaults-file=/root/.my.cnf -e "grant all on nextcloud.* to 'nextcloud'@'localhost' identified by '$password';"
-/usr/bin/mysql --defaults-file=/root/.my.cnf -e "FLUSH PRIVILEGES"
+systemctl is-active --quiet rh-mariadb105-mariadb@nextcloud
+if [[ $? -ne 0 ]]; then
+    systemctl start rh-mariadb105-mariadb@nextcloud
+    max_wait=20
+    wait=0
+    while ! mysql --socket /run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "SHOW DATABASES" 2>/dev/null; do
+        sleep 1
+        wait=$((wait+1))
+        if [ $wait -ge $max_wait ]; then
+            echo "[ERROR] Can't start rh-mariadb105-mariadb@nextcloud"
+            exit 1
+        fi
+    done
+fi
+
+if [[ -d '/var/lib/mysql/nextcloud' ]]; then
```

Comment:
Can we add also a query to mysql to see if `nextcloud` db exists?
I fear the directory could not be completely removed if the admin has enabled slow query log or similar.

Reply:
sorry I forgot this one lets go on it

Reply:
should be good @gsanchietti